src/crimson/os/seastore/object_data_handler.h

```diff
@@ -35,24 +51,23 @@ struct ObjectDataBlock : crimson::os::seastore::LogicalCachedExtent {
     return TYPE;
   }
 
-  ceph::bufferlist get_delta() final {
-    /* Currently, we always allocate fresh ObjectDataBlock's rather than
-     * mutating existing ones. */
-    ceph_assert(0 == "Should be impossible");
+  void add_contents(bufferlist bl, uint64_t offset) {
```

Comment:
How about renaming this to `overwrite(extent_len_t offset, bufferlist bl)`, and do overwrite extent-bl and append delta together here ?

Proper asserts can also be added, e.g. the overwrite range should not larger than the extent.

Reply:
Done.